pkg/rpc/client/rpc.go

```diff
@@ -588,3 +598,42 @@ func (c *Client) CalculateValidUntilBlock() (uint32, error) {
 	}
 	return blockCount + validatorsCount, nil
 }
+
+// AddNetworkFee adds network fee for each witness script to transaction.
+func (c *Client) AddNetworkFee(tx *transaction.Transaction, acc *wallet.Account) error {
+	size := io.GetVarSize(tx)
+	if acc.Contract != nil {
+		tx.NetworkFee += calculateNetworkFee(acc.Contract.Script, &size)
+	}
+	for _, cosigner := range tx.Cosigners {
+		contract, err := c.GetContractState(cosigner.Account)
+		if err != nil {
+			return err
+		}
+		tx.NetworkFee += calculateNetworkFee(contract.Script, &size)
+	}
+	//TODO:
+	//tx.NetworkFee += size * NativeContract.Policy.GetFeePerByte;
+	return nil
+}
+
+func calculateNetworkFee(script []byte, size *int) util.Fixed8 {
+	var netFee util.Fixed8
+	if vm.IsSignatureContract(script) {
+		*size += 67 + io.GetVarSize(script)
+		netFee += opcode.GetPrice(opcode.PUSHDATA1, opcode.PUSHDATA1, opcode.PUSHNULL) // TODO + price of Neo.Crypto.ECDsaVerify
+	} else if m, n, _, ok := vm.ParseMultiSigContract(script); ok {
+		sizeInv := 66 * m
+		*size += io.GetVarSize(sizeInv) + sizeInv + io.GetVarSize(script)
+		netFee += util.Fixed8FromInt64(opcode.GetPrice(opcode.PUSHDATA1).IntegralValue() * int64(m))
+		bw := io.NewBufBinWriter()
+		emit.Int(bw.BinWriter, int64(m))
+		netFee += opcode.GetPrice(opcode.Opcode(bw.Bytes()[0]))
```

Comment:
This line and 2 ones above deserve a separate function.

Reply:
I also think this is invalid way to do price calculation. It assumes everything was generated by our compiler. `0` could be emitted by each of `PUSH0`, `PUSHINT8`, `PUSHINT16`. This line calculates correct price just because  prices of small `PUSH*` opcodes are equal (what a coincidence!). We can easily replace it with constant then.

Reply:
Move this method to core/util.go. So, should we replace `GetPrice` with `30`?

Reply:
No, but may be drop some comment, not to forget about this.